File: frontend/EduMatch/src/types/session.ts
export type SessionStatus = "ready" | "active" | "completed" | "cancelled" | "disputed";

export interface Session {
  id: number;
  request_id: number;
  student_id: number;
  instructor_id: number;
  session_mode: string;
  session_type: string;
  scheduled_at: string | null;
  started_at: string | null;
  ended_at: string | null;
  instructor_marked_completed_at: string | null;
  student_confirmed_completed_at: string | null;
  completed_at: string | null;
  status: SessionStatus;
  created_at: string;
  updated_at: string;
  request_title: string | null;
  request_status: string | null;
  payment_status: string | null;
  payment_amount: string | null;
  payment_platform_fee: string | null;
  payment_total_amount: string | null;
  student_name: string | null;
  instructor_name: string | null;
  has_review: boolean;
}

File: backend/app/models/session.py
from datetime import datetime

from sqlalchemy import DateTime, ForeignKey, String, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database import Base


class Session(Base):
    __tablename__ = "sessions"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    request_id: Mapped[int] = mapped_column(ForeignKey("requests.id", ondelete="CASCADE"), index=True, nullable=False)
    student_id: Mapped[int] = mapped_column(ForeignKey("users.id", ondelete="CASCADE"), index=True, nullable=False)
    instructor_id: Mapped[int] = mapped_column(ForeignKey("users.id", ondelete="CASCADE"), index=True, nullable=False)
    session_mode: Mapped[str] = mapped_column(String(32), nullable=False, default="individual", server_default="individual")
    session_type: Mapped[str] = mapped_column(String(32), nullable=False, default="online", server_default="online")
    scheduled_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    started_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    ended_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    instructor_marked_completed_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    student_confirmed_completed_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    completed_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    status: Mapped[str] = mapped_column(String(32), nullable=False, default="ready", server_default="ready")
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
        nullable=False,
    )

    request = relationship("LearningRequest", back_populates="sessions")
    student = relationship("User", foreign_keys=[student_id], back_populates="student_sessions")
    instructor = relationship("User", foreign_keys=[instructor_id], back_populates="instructor_sessions")
    payments = relationship("Payment", back_populates="session", cascade="all, delete-orphan")
    messages = relationship("Message", back_populates="session", cascade="all, delete-orphan")
    reviews = relationship("Review", back_populates="session", cascade="all, delete-orphan")
